go/cli/mcap/cmd/info.go

```diff
@@ -45,16 +45,23 @@ func printInfo(w io.Writer, info *mcap.Info) error {
 		fmt.Fprintf(buf, "messages: %d\n", info.Statistics.MessageCount)
 		start = info.Statistics.MessageStartTime
 		end = info.Statistics.MessageEndTime
+		durationNs := float64(end) - float64(start)
+		durationInSeconds = durationNs / 1e9
```

Comment:
Can we do the subtraction before converting to float? That would be less likely to lose precision due to catastrophic cancellation.

Reply:
yes, but it's a little more complex - we have to make sure the unsigned subtraction won't overflow first.

Reply:
sounds good...maybe an error message on negative duration? or do we want to actually display it with a `-` sign?